classes/01class/exercises/c01-aws01/gpavelar/README.md

```diff
@@ -0,0 +1,436 @@
+              "SubnetId": "subnet-351ced53",
+              "VpcId": "vpc-37d4dd50",
+              "InterfaceType": "interface"
+            }
+          ],
+          "RootDeviceName": "/dev/xvda",
+          "RootDeviceType": "ebs",
+          "SecurityGroups": [
+            {
+              "GroupName": "devops-develop",
+              "GroupId": "sg-076a9212c8d1a8a45"
+            }
+          ],
+          "SourceDestCheck": true,
+          "VirtualizationType": "hvm",
+          "CpuOptions": {
+            "CoreCount": 1,
+            "ThreadsPerCore": 1
+          },
+          "CapacityReservationSpecification": {
+            "CapacityReservationPreference": "open"
+          },
+          "HibernationOptions": {
+            "Configured": false
+          },
+          "MetadataOptions": {
+            "State": "applied",
+            "HttpTokens": "optional",
+            "HttpPutResponseHopLimit": 1,
+            "HttpEndpoint": "enabled"
+          }
+        }
+      ],
+      "OwnerId": "478433196210",
+      "ReservationId": "r-0927dd5e49e7f6d40"
+    }
+  ]
+}
+```
+
+```bash
+## Private IP - 172.31.3.15
+ssh -i /var/scratch/DevOps_Academy/DevOps-academy.pem  ec2-user@172.31.2.153
+ssh: connect to host 172.31.2.153 port 22: Connection timed out
+```
+
+- It's not possible to connect via SSH, because it is not available to be accessed.
+
+So we will overcome this step using [SSH Agent](https://aws.amazon.com/blogs/security/securely-connect-to-linux-instances-running-in-a-private-amazon-vpc/). But we need to adapt the command because the OS ([ubuntu command here](https://blog.earth-works.com/2015/01/29/add-an-ec2-pem-key-to-ssh-on-linux-and-mac-os-x/))
+
+```bash
+ssh-add /var/scratch/DevOps_Academy/DevOps-academy.pem
+
+ssh -A ec2-user@3.25.65.76
+Last login: Fri Jun 19 09:57:06 2020 from 147.10.73.48
+
+       __|  __|_  )
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+4 package(s) needed for security, out of 10 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-0-98 ~]$
+
+# Now accessing the instance that we want
+[ec2-user@ip-172-31-0-98 ~]$ ssh ec2-user@172.31.2.153
+The authenticity of host '172.31.2.153 (172.31.2.153)' can't be established.
+ECDSA key fingerprint is SHA256:fd2nMFYkmUa/hRnBenth4SdIMtFIOAJ+2KN/mbVa6m8.
+ECDSA key fingerprint is MD5:da:b1:8f:a2:98:1b:53:9b:aa:17:72:cc:d2:f4:57:b3.
+Are you sure you want to continue connecting (yes/no)? yes
+Warning: Permanently added '172.31.2.153' (ECDSA) to the list of known hosts.
+
+       __|  __|_  )
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+[ec2-user@ip-172-31-2-153 ~]$
+
+
+```
+
+**PS**: I ran the command to create instance many times, and I was checking it on amazon console and none of those instance appeared. I realized that I was checking the wrong region... "/. There were 10 instances running hahahhahaha.
```

Comment:
WOW that's dangerous ($$$).

set the default region in the CLI using `aws configure`.

Reply:
Now I'm paying attention to this. hahaha I had set up aws configure, But I was listing the wrong one